modules/tensor_mechanics/include/userobjects/TensorMechanicsPlasticDruckerPrager.h

```diff
@@ -0,0 +1,181 @@
+  /**
+   * The derivative of yield function with respect to stress
+   * @param stress the stress at which to calculate the yield function
+   * @param intnl internal parameter
+   * @return df_dstress(i, j) = dyieldFunction/dstress(i, j)
+   */
+  virtual RankTwoTensor dyieldFunction_dstress(const RankTwoTensor & stress, Real intnl) const;
+
+  /**
+   * The derivative of yield function with respect to the internal parameter
+   * @param stress the stress at which to calculate the yield function
+   * @param intnl internal parameter
+   * @return the derivative
+   */
+  virtual Real dyieldFunction_dintnl(const RankTwoTensor & stress, Real intnl) const;
+
+  /**
+   * The flow potential
+   * @param stress the stress at which to calculate the flow potential
+   * @param intnl internal parameter
+   * @return the flow potential
+   */
+  virtual RankTwoTensor flowPotential(const RankTwoTensor & stress, Real intnl) const;
+
+  /**
+   * The derivative of the flow potential with respect to stress
+   * @param stress the stress at which to calculate the flow potential
+   * @param intnl internal parameter
+   * @return dr_dstress(i, j, k, l) = dr(i, j)/dstress(k, l)
+   */
+  virtual RankFourTensor dflowPotential_dstress(const RankTwoTensor & stress, Real intnl) const;
+
+  /**
+   * The derivative of the flow potential with respect to the internal parameter
+   * @param stress the stress at which to calculate the flow potential
+   * @param intnl internal parameter
+   * @return dr_dintnl(i, j) = dr(i, j)/dintnl
+   */
+  virtual RankTwoTensor dflowPotential_dintnl(const RankTwoTensor & stress, Real intnl) const;
+
+  /**
+   * The parameters aaa and bbb are chosen to closely match the Mohr-Coulomb
+   * yield surface.  Various matching schemes may be used and this parameter
+   * holds the user's choice.
+   */
+  const MooseEnum _mc_interpolation_scheme;
+
+  /// True if there is no hardening of cohesion
+  const bool _zero_cohesion_hardening;
+
+  /// True if there is no hardening of friction angle
+  const bool _zero_phi_hardening;
+
+  /// True if there is no hardening of dilation angle
+  const bool _zero_psi_hardening;
+
+  /// Calculates aaa and bbb as a function of the internal parameter intnl
+  void AandB(Real intnl, Real & aaa, Real & bbb) const;
+
+  /// Calculates d(aaa)/d(intnl) and d(bbb)/d(intnl) as a function of the internal parameter intnl
+  void dAandB(Real intnl, Real & daaa, Real & dbbb) const;
+
+  /**
+   * bbb (friction) and bbb_flow (dilation) are computed using the same function,
+   * Bonly, and this parameter tells that function whether to compute bbb or bbb_flow
+   */
+  enum fric_or_dil { friction = 0, dilation = 1 };
+
+  /**
+   * Calculate bbb or bbb_flow
+   * @param intnl the internal parameter
+   * @param fd if fd==friction then bbb is calculated.  if fd==dilation then bbb_flow is calculated
+   * @param bbb either bbb or bbb_flow, depending on fd
+   */
+  void Bonly(Real intnl, int fd, Real & bbb) const;
```

Comment:
this should be `bOnly` if we are strict about our code conventions...

Similarly the above should be `aAndB` (yes the capitalization does not look as nice, but maybe the non-descript name "a and b" is the real problem).

Also the enum should be named like a type (capitalized CamelCase), i.e. `FricOrDil` (and preferable `FrictionDialation`, to avoid the awkward abbreviations).